tests/gdb-tests/tests/test_gdblib_parameter.py

```diff
@@ -6,31 +6,57 @@
 
 @pytest.mark.parametrize(
     "params",
-    (("int", 123, "123"), ("bool", True, "on"), ("string", "some-string-val", "some-string-val")),
+    (
+        ("int", 123, "123", {}),
+        ("bool", True, "on", {}),
+        ("bool", False, "off", {}),
+        ("string", "some-string-val", "some-string-val", {}),
+        ("auto-bool", None, "auto", {"param_class": gdb.PARAM_AUTO_BOOLEAN}),
+        ("unlimited-uint", 0, "unlimited", {"param_class": gdb.PARAM_UINTEGER}),
+        ("unlimited-int", 0, "unlimited", {"param_class": gdb.PARAM_INTEGER}),
+        ("unlimited-zuint", -1, "unlimited", {"param_class": gdb.PARAM_ZUINTEGER_UNLIMITED}),
+        (
+            "enum",
+            "enum1",
+            "enum1",
+            {"param_class": gdb.PARAM_ENUM, "enum_sequence": ["enum1", "enum2", "enum3"]},
+        ),
+    ),
 )
 def test_gdb_parameter_default_value_works(start_binary, params):
-    name_suffix, default_value, displayed_value = params
+    name_suffix, default_value, displayed_value, optional_kwargs = params
 
     param_name = f"test-param-{name_suffix}"
+    help_docstring = f"Help docstring for {param_name}"
+    optional_kwargs["help_docstring"] = help_docstring
```

Comment:
This seems to be the only keyword arg, why do we need `optional_kwargs`? We can just do `help_docstring=help_docstring` in the call below.

Reply:
> This seems to be the only keyword arg, why do we need `optional_kwargs`?

`optional_kwargs` is for testing the auto boolean and other parameter classes when we are not using the default `param_class`.

For example, when we are testing the `gdb.PARAM_ENUM` parameter, `optional_kwargs` will be:

```python
{"param_class": gdb.PARAM_ENUM, "enum_sequence": ["enum1", "enum2", "enum3"]}
```

> We can just do help_docstring=help_docstring in the call below.

Yes, this looks better, I'll pass `help_docstring` directly.